geometry/sign.hpp

```diff
@@ -4,41 +4,65 @@
 #include <string>
 
 #include "base/not_null.hpp"
+#include "quantities/quantities.hpp"
 #include "serialization/geometry.pb.h"
 
 namespace principia {
 namespace geometry {
 namespace internal_sign {
 
 using base::not_null;
+using quantities::Quantity;
 
 // An element of the multiplicative group ({+1, -1}, *). Useful for instance to
 // represent the determinant of an orthogonal map.
 class Sign final {
  public:
-  template<typename Scalar> explicit Sign(Scalar const& s);
+  explicit Sign(double x);
+  template<typename Dimensions>
+  explicit Sign(Quantity<Dimensions> const& x);
+  // The deleted constructor forbids construction from an integer via
+  // integer-to-double conversion.  Integers have no signed 0, so this could
+  // lead to confusing behaviour.
+  template<typename T>
+  explicit Sign(T x) = delete;
+
+  static constexpr Sign Positive();
+  static constexpr Sign Negative();
+
+  template<typename T, typename = std::enable_if_t<std::is_arithmetic_v<T>>>
```

Comment:
I'd prefer `is_integral` here.  It's quite weird to call this function with a floating-point type when you have perfectly good constructors to do that.